compiler/il/OMRDataTypes.cpp

```diff
@@ -318,3 +318,27 @@ OMR::DataType::getPrefix(TR::DataType dt)
    TR_ASSERT(dt < TR::NumOMRTypes, "Prefix requested for unknown datatype");
    return OMRDataTypePrefixes[dt];
    }
+
+float TR::getMaxFloat()
+   {
+   int32_t f = FLOAT_POS_INFINITY;
+   return *(float*)&f;
```

Comment:
This kind of cast relies on undefined behaviour because it violates strict aliasing.

Reply:
It is also possible to use `reinterpret_cast` to interpret the underlining bit pattern as a floating point type[1]. Do you have any suggestion on which method I should use, or do you have better way of doing this?
[1] 
```
// x is a pointer to float inf here
int f = FLOAT_POS_INFINITY;
float *x = reinterpret_cast<float*> (&f);
```

Reply:
An explicit `reinterpret_cast` doesn't change anything (the c-style cast is really just a hidden `reinterpret_cast`). The only safe way of doing this in C++ is to `memcpy` the int to a temporary float [1,2]:

[1] https://en.cppreference.com/w/cpp/string/byte/memcpy
[2]
```c++
int32_t f = FLOAT_POS_INFINITY;
float x = 0.0;
std::memcpy(&x, &f, sizeof(f));
```

Reply:
I have used memcpy in the new commit f0a3ce4